### PR TITLE
Windows

### PR DESCRIPTION
**Pull request by richardschneider**
Get it working on windows.

Note that I replaced the package `pull-glob` with `glob`; because it does work on windows.

This PR is dependent on https://github.com/ipfs/interface-datastore/pull/12 getting released.

### CODE REVIEW

**Comment by richardschneider** [comment]
Waiting on https://github.com/ipfs/interface-datastore/pull/12 and https://github.com/ipfs/js-datastore-core/pull/3 to be released.

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "src/index.js",
   "scripts": {
     "lint": "aegir-lint",
-    "test": "aegir-test --env node",
+    "test": "aegir-test --env node --timeout 20000",
```
Comment (dryajov) [inline package.json]: Why do we need this now? Have timeouts increased with this changes? If so, we should have individual timeouts.

File: package.json [hunk 2]
```diff
@@ -55,6 +55,7 @@
   },
   "contributors": [
     "David Dias <daviddias.p@gmail.com>",
-    "Friedel Ziegelmayer <dignifiedquire@gmail.com>"
+    "Friedel Ziegelmayer <dignifiedquire@gmail.com>",
+    "Richard Schneider <makaretu@gmail.com>"
```
Comment (dryajov) [inline package.json]: this is not needed, its added automatically on release.

File: src/index.js [hunk 3]
```diff
@@ -259,7 +261,13 @@ class FsDatastore {
    * @returns {PullStream}
    */
   query (q /* : Query<Buffer> */) /* : QueryResult<Buffer> */ {
-    let tasks = [glob(path.join(this.path, '**', '*' + this.opts.extension))]
+    // glob expects a POSIX path
+    let prefix = q.prefix || '**'
+    let pattern =
+      path.join(this.path, prefix, '*' + this.opts.extension)
+      .split(path.sep)
+      .join('/')
```
Comment (dryajov) [inline src/index.js]: shouldn't this be the inverse? Split by the key separator `/` and the join with the os specific path sep?

File: package.json (hunk 1 above)
Comment (richardschneider) [reply package.json]: Yes a lot of tests were skipped.  In particular the `many (3 * 400)` test takes a long time on my top-end laptop when sharding.

File: package.json (hunk 2 above)
Comment (richardschneider) [reply package.json]: ok will reset

File: src/index.js (hunk 3 above)
Comment (richardschneider) [reply src/index.js]: Beg to differ.  The `path.join` creates an os specific path. The `split(path.sep)` then gives an array from the os specific and then `join('/')` creates a POSIX specific path, which is what `glob` wants,

Comment (dryajov) [reply src/index.js]: ah `glob.sync` does expect POSIX [`/`](https://github.com/isaacs/node-glob#windows]) separators - makes sense 👍

**Comment by richardschneider** [comment]
@dryajov do you know about `flow` and why the Travis CI [build](https://travis-ci.org/ipfs/js-datastore-core/jobs/296496462) is failing?

**Comment by dryajov** [comment]
yeah, flow will enable some stricter type checking - looking at travis now.

**Comment by richardschneider** [comment]
Why is `flow` complaining about code in `node-modules`?

**Comment by dryajov** [comment]
@richardschneider I believe this might help - https://github.com/ipfs/js-datastore-fs/pull/5. Could you try those changes in your branch?

**Comment by dryajov** [comment]
@richardschneider you also need those `type-flow` files. I would rebase my commit onto your branch.

**Comment by dryajov** [comment]
@dignifiedquire do we still want to use flow? I haven't seen it used anywhere else in the codebase  and it might not worth the effort to support it at this point. _I'm not sure why it broke now - though, that's a good question._

**Comment by dignifiedquire** [comment]
Yes, unless there is a strong reason to remove it I believe it is good to keep it as it makes sure that the `interface-datastore` contracts are not accidentally broken.

**Pull request merged**